keps/sig-cli/0014-krew.md

```diff
@@ -0,0 +1,237 @@
+    * [Lifecycle management](#lifecycle-management)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+      * [Capabilities](#capabilities)
+    * [Scenarios (standalone only)](#scenarios-standalone-only)
+    * [Scenarios](#scenarios)
+    * [UX and CLI Flow](#ux-and-cli-flow)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints-optional)
+    * [Risks](#risks)
+  * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+  * [Drawbacks](#drawbacks)
+
+## Summary
+
+krew is a package manager for the kubectl plugin system. krew introduces a way
+to handle kubectl plugin management and discovery.
+
+## Motivation
+
+The kubectl plugin system allows users to extend kubectl by placing extension
+executables in a plugin directory. This allows new subcommands and subcommand
+trees to be added in the kubectl CLI interface under `kubectl plugin <NAME>`.
+This has the following user friction:
+
+* Users need to manually download and manage plugins to ~/.kube/plugins and
+  keep them up to date
+* Users need to be aware of the installation requirements and instructions of
+  each plugin
+
+Currently, there are no unified distribution methods for kubectl plugins.
+This has several disadvantages:
+
+### Discoverability
+
+Available kubectl plugins are not easily discoverable and searchable.
+
+### Cross-platform Packaging
+
+Plugins need to be packaged separately different OS and distros.
+Often, plugin maintainers do not do this.
+
+### Lifecycle management 
+
+Often installation is done through manually downloading a zip and placing
+it under ~/.kube/plugins. This is an unpleasant experience to the user
+especially when it comes to updating the plugins,
+or when a user deletes ~/.kube.
+
+---
+
+Those disadvantages will lead to slow the adoption rate of kubectl plugins.
+krew will be a plugin manager able to solve all of the named issues. 
+
+### Goals
+
+
+* Install, uninstall and update plugins
+* Discovery of plugins
+* Giving developers a platform to contribute their plugins to
+* Build repository platform for contributors (i.e brew formulas approach)
+* Allow multiple plugin index sources
+* (only for standalone) Self hosting: krew installation doesn't require a
+  package manager: it can upgrade itself 
+
+### Non-Goals
+
+* Dependency management of libraries
+* Plugins are user facing programs, they should not depend on each other.
+  They have environment requirements like curl or git.
+  Krew is not responsible to cover external dependency,
+  program and environment setup like apt,
+  brew or dnf. The user will be informed to take action.
+* Complete integration with various download sources like “hg://” or “samba://”
+* kubectl version compatibility management
+
+
+## Proposal
+
+#### Capabilities
+
+Note: This proposal has already been implemented in
+`github.com/GoogleContainerTools/krew` as a proof of concept.
```

Comment:
Does not exist.

Reply:
The repository is now publicly accessible 😄 
Same URL: https://github.com/GoogleContainerTools/krew